### PR TITLE
util/syspolicy: finish plumbing policyclient, add feature/syspolicy, move global impl

### PR DESCRIPTION
**Pull request by bradfitz**
This is step 4 of making syspolicy a build-time feature.

This adds a policyclient.Get() accessor to return the correct
implementation to use: either the real one, or the no-op one. (A third
type, a static one for testing, also exists, so in general a
policyclient.Client should be plumbed around and not always fetched
via policyclient.Get whenever possible, especially if tests need to use
alternate syspolicy)

Updates #16998
Updates #12614

### CODE REVIEW

File: util/syspolicy/syspolicy.go [hunk 1]
```diff
@@ -237,3 +231,54 @@ func SelectControlURL(reg, disk string) string {
...
+func (globalSyspolicy) GetPreferenceOption(name pkey.Key, defaultValue ptype.PreferenceOption) (ptype.PreferenceOption, error) {
+	return getPreferenceOption(name, defaultValue)
+}
+
+func (globalSyspolicy) GetVisibility(name pkey.Key) (ptype.Visibility, error) {
+	return getVisibility(name)
+}
+
+func (globalSyspolicy) HasAnyOf(keys ...pkey.Key) (bool, error) {
+	return HasAnyOf(keys...)
```
Comment (nickkhyl) [inline util/syspolicy/syspolicy.go]: nit: unexport the implementation? It's a bit weird that everything is now unexported except for `syspolicy.SetDebugLoggingEnabled` and `syspolicy.HasAnyOf`.

File: util/syspolicy/syspolicy.go [hunk 2]
```diff
@@ -6,8 +6,8 @@
 // changes.
 //
 // It provides functions that return specific policy settings by their unique
-// [setting.Key]s, such as [GetBoolean], [GetUint64], [GetString],
-// [GetStringArray], [GetPreferenceOption], [GetVisibility] and [GetDuration].
+// [setting.Key]s, such as [getBoolean], [getUint64], [getString],
+// [getStringArray], [getPreferenceOption], [getVisibility] and [getDuration].
```
Comment (nickkhyl) [inline util/syspolicy/syspolicy.go]: Update the package doc to say it registers a `policyclient.Client` when imported instead of listing now-unexported funcs?

File: util/syspolicy/policyclient/policyclient.go [hunk 3]
```diff
@@ -66,6 +66,21 @@ type Client interface {
 	RegisterChangeCallback(cb func(PolicyChange)) (unregister func(), err error)
 }
 
+// Get returns a non-nil [Client] implementation as a function of the
+// build tags. It returns a no-op implementation if the full syspolicy
+// package is omitted from the build.
+func Get() Client {
```
Comment (nickkhyl) [inline util/syspolicy/policyclient/policyclient.go:72]: I get why we currently need `policyclient.Get` alongside `(*tsd.System).PolicyClientOrDefault`. Is the plan to plumb `tsd.System` everywhere and eventually drop `policyclient.Get`? Or do we foresee using both in the long term?

**Review by nickkhyl — APPROVED** [review]
Thanks!

File: util/syspolicy/syspolicy.go (hunk 1 above)
Comment (bradfitz) [reply util/syspolicy/syspolicy.go]: Whoops, done. I just missed those.

File: util/syspolicy/syspolicy.go (hunk 2 above)
Comment (bradfitz) [reply util/syspolicy/syspolicy.go]: done

File: util/syspolicy/policyclient/policyclient.go (hunk 3 above)
Comment (bradfitz) [reply util/syspolicy/policyclient/policyclient.go:72]: No, we can't plumb tsd.System everywhere as it pulls in the whole universe. That's really only for LocalBackend, which is the ~only thing that takes a tsd.System.

The idea is that tsd.System should be populated with an alternate policy client if you really need to (for testing) but otherwise the empty value is fine, to get the reasonable default of real-vs-noop.

And then wherever that can take/plumb the policyclient should.

But if you really can't, then policyclient.Get() is your fallback. Notably for all the things that previously used globals.

Of the ones that remain, I might be able to get rid of maybe half of them with some effort, but none seem worth it.
